### PR TITLE
Emit 'change' events in 'set' when nested object attributes have chan…

### PR DESCRIPTION
**Pull request by pricj004**
…ged, and root-level scalars with identical values are present

cc @chuym

### CODE REVIEW

File: spec/ObjectUtilsSpec.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
+import ObjectUtils from '../src/ObjectUtils';
+
+describe('ObjectUtils', () => {
```
Comment (bradvogel) [inline spec/ObjectUtilsSpec.js]: awesome!

File: src/ObjectUtils.js [hunk 2]
```diff
@@ -81,6 +81,38 @@ var ObjectUtils = {
...
+        clone[k] = ObjectUtils.deepClone(v);
+      }
+    });
+    return clone;
+  },
+
+  /**
+   * Returns the changes between two objects. Includes attributes in `b` that are not in `a`, and
+   * attributes that are in both `a` and `b` but where the values differ. Performs deep comparison
+   * of objects and arrays.
```
Comment (ttacon) [inline src/ObjectUtils.js]: Can we clarify the semantics here around arrays? It's somewhat ambiguous at the moment if the code should return the full array when values change, or simple the values in the array that are changed (the first is the current behaviour as per this [test](https://github.com/mixmaxhq/backbone-publication/pull/11/files#diff-9eccd12b5bb0d9ee95122aeb932a308aR187)).

File: src/PublicationModel.js [hunk 3]
```diff
@@ -69,30 +69,37 @@ var PublicationModel = Backbone.Model.extend({
...
+
+    // If there were no real changes, we can bail.
+    var hasChanges = !_.isEmpty(changedAttributes);
+    if (!hasChanges) return;
+
+    /* Perform the standard Backbone.js set. Do this first, so when we trigger events further down
+     * the new attributes are in place. We also determine if all the changes were to nested
+     * objects - if so, we don't emit any events (but still call `set` so the new attributes are
+     * stored on the model).  */
+    var allChangesAreObjects = _.every(changedAttributes, v => _.isObject(v));
```
Comment (ttacon) [inline src/PublicationModel.js]: nit: Let's wrap the params for the fat arrow function in parens for readability (so `_.every(changedAttributes, (v) => _.isObject(v));`).

Comment (wearhere) [inline src/PublicationModel.js]: If you liked you could also do `_.every(changedAttributes, _.isObject)`.

**Pull request merged**